### PR TITLE
WIP: #241 shouldi got Cargo-audit for RUST code

### PR DESCRIPTION
**Pull request by Yash-Varshney**
added cargo_audit.py and test_cargo_audit.py in examples/shouldi/shouldi and examples/shouldi/tests respectively.

### CODE REVIEW

**Comment by Yash-Varshney** [comment]
@pdxjohnny  pls review !!!

File: examples/shouldi/tests/test_cargo_audit.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+from dffml.util.asynctestcase import AsyncTestCase
+
+from shouldi.cargo_audit import run_cargo_audit
+
+
+class TestRunCargo_AuditOp(AsyncTestCase):
+    @cached_download_unpack_archive(
+        "https://static.rust-lang.org/dist/rust-1.42.0-x86_64-unknown-linux-gnu.tar.gz",
+        pathlib.Path(__file__).parent / "cargo_audit.tar.gz",
+        pathlib.Path(__file__).parent / "cargo-audit-download",
```
Comment (pdxjohnny) [inline examples/shouldi/tests/test_cargo_audit.py]: It looks like you've got the downloads for rust and tarpaulin, but you've named the rust download `cargo_audit` and don't have the download for `cargo_audit` itself.

File: examples/shouldi/tests/test_cargo_audit.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        pathlib.Path(__file__).parent / "tarpaulin.tar.gz",
+        pathlib.Path(__file__).parent / "tarpaulin-download",
+        "03c10779bdf09baa9b6a8caab367de8e780e9b72778b40c017adb85c1b1ec38b96355ba67482067fcd06917632a81f69",
+    )
+    async def test_run(self, cargo_audit, tarpaulin):
+        with prepend_to_path(cargo_audit / "bin",):
+            results = await run_cargo_audit(
+                str(
+                    tarpaulin
+                    / "tarpaulin-59f1c4f48765fba27319cf64e8aab7b08f8a0f66 (1)"
```
Comment (pdxjohnny) [inline examples/shouldi/tests/test_cargo_audit.py]:
```suggestion
                    / "tarpaulin-59f1c4f48765fba27319cf64e8aab7b08f8a0f66"
```

File: examples/shouldi/.gitignore [hunk 3]
```diff
@@ -21,5 +21,7 @@ htmlcov/
 tests/golangci-lint-download/
 tests/javascript_algo-download/
 tests/npm-audit-download/
+tests/cargo-audit-download/
+tests/tarpaulin-download/
```
Comment (pdxjohnny) [inline examples/shouldi/.gitignore]: If you could make another PR which puts all of these downloads under `tests/downloads` that would be good, so for example in that PR this line would be `tests/downloads/tarpaulin`

File: CHANGELOG.md [hunk 4]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Shouldi got an operation to run npm-audit on JavaScript code
 - Docstrings and doctestable examples for `record.py` (features and evaluated)
 - Simplified model API with SimpleModel
+- Shouldi got an operation to run cargo-aduit on RUST code.
```
Comment (pdxjohnny) [inline CHANGELOG.md]:
```suggestion
- shouldi got an operation to run cargo-aduit on rust code.
```

I don't think we want capitalization on `shouldi` or `rust`

File: examples/shouldi/.gitignore (hunk 3 above)
Comment (pdxjohnny) [reply examples/shouldi/.gitignore]: This will allow us to put the downloads in the cache to speed up the CI tests. You will of course also need to change the paths in the calls to `cached_download_unpack_archive`

Comment (Yash-Varshney) [reply examples/shouldi/.gitignore]: sorry for being late, was stuck btw things. Do I need to add all the previous downloads in this file as under `tests/downloads` or only the new ones.

**Comment by pdxjohnny** [comment]
Closed because of https://github.com/intel/dffml/pull/493
